### ISSUE TITLE
v3.0.0-beta.16 useQuery returns stale data when in same component as useMutation

### ISSUE THREAD

**Issue by cbergmiller** [apollographql/apollo-client#5717]
In an CRUD-Component  i have `useQuery` and `useMutation` hooks:
```tsx
const {data: locationData} = useQuery(ListLocations, {variables: {siteId}});
const [locationForm, {loading: isSubmitting}] = useMutation(LocationFormMutation);
```
**Intended outcome:**
After executing the mutation (updating a _location_ entity) the cache should update and useQuery should return a new list of _location_ records with the updated _location_ entity.

**Actual outcome:**
The cache is updated. The query data is not. When i insert the exact same query hook in a parent component, the returned data is updated immediately after the mutation.

**Versions**
"@apollo/client": "3.0.0-beta.16",
"apollo-link-error": "2.0.0-beta.0",

**Comment by cbergmiller** [comment]
I did some investigating.
In **QueryData.ts** the observable subscription `next` method is called after the mutation executes.
`onNewData` is also called but the `tick` reducer in the `useBaseQuery` does not increment.
I checked that my Component is not remounting (it isn't).
It is however wrapped in ReactRouter, and other contexts.

**Comment by cbergmiller** [comment]
After removing Strict-Mode (`<React.StrictMode>`) from my App,the useQuery hook now is updating as expected. I don't know if this is a bug in React, in Apollo-Client or in my App. Feel free to close the issue or to request additional investigation on my side.

**Comment by benjamn** [comment]
While we've been focusing primarily on the cache in Apollo Client 3.0, we are aware that not every change to the cache automatically/immediately propagates back through the client and into React Apollo. This is definitely an area for improvement before AC3 is finalized, since it's an important part of the reactivity story.

Thanks for helping test the betas!

**Comment by hwillson** [comment]
@cbergmiller any chance you'd be able to test with the latest `@apollo/client@beta` (`3.0.0-beta.24`) with `StrictMode`, and let us know if this is still a problem? Thanks!

**Comment by cbergmiller** [comment]
@hwillson The problem is fixed now (tested with 3.0.0-beta.24). Thanks to the Apollo team!

**Comment by hwillson** [comment]
Great to hear @cbergmiller - thanks!
